File: aura-ecommerce-api/src/main/java/orgcom/auraecommerceapi/services/fasad/UserManagementService.java
package orgcom.auraecommerceapi.services.fasad;

import orgcom.auraecommerceapi.security.entities.User;
import orgcom.auraecommerceapi.shared.ResponseGenericResult;

import java.util.List;

public interface UserManagementService {
    List<User> getAllUsers();
    ResponseGenericResult<Boolean> saveUser(User user);
    ResponseGenericResult<Boolean> deleteUser(Long id);
    ResponseGenericResult<Boolean> updateUser(User user);
}

File: aura-ecommerce-ui/src/app/aura-ecommerce/components/pages/users/services/user-management.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { AuraUser } from '../models/aura-user.model';
import { Observable } from 'rxjs';
import { environment } from 'src/environments/environment';
import { ResponseGenericResult } from 'src/app/shared/models/response-generic-result/response-generic-result.model';

@Injectable({
  providedIn: 'root'
})
export class UserManagementService {

  private baseUrl = environment.apiUrl +'/users';

  constructor(private http: HttpClient) { }

  getAllUsers(): Observable<AuraUser[]> {
    return this.http.get<AuraUser[]>(this.baseUrl);
  }

  createUser(user: AuraUser): Observable<ResponseGenericResult<boolean>> {
    return this.http.post<ResponseGenericResult<boolean>>(this.baseUrl, user);
  }

  updateUser(user: AuraUser): Observable<ResponseGenericResult<boolean>> {
    return this.http.put<ResponseGenericResult<boolean>>(`${this.baseUrl}`, user);
  }

  deleteUser(id: number): Observable<ResponseGenericResult<boolean>> {
    return this.http.delete<ResponseGenericResult<boolean>>(`${this.baseUrl}/${id}`);
  }}
